pkg/registry/registry_aliyun.go

```diff
@@ -0,0 +1,118 @@
+package registry
+
+import (
+	"bytes"
+	"encoding/json"
+	"strings"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cr"
+	"github.com/giantswarm/microerror"
+)
+
+type AliyunTag struct {
+	ImageUpdate int64  `json:"imageUpdate"`
+	ImageCreate int64  `json:"imageCreate"`
+	ImageID     string `json:"imageId"`
+	Digest      string `json:"digest"`
+	ImageSize   int64  `json:"imageSize"`
+	Tag         string `json:"tag"`
+	Status      string `json:"status"`
+}
+
+type AliyunResponseWrapper struct {
+	Data AliyunTagsResponse `json:"data"`
+}
+
+type AliyunTagsResponse struct {
+	Tags         []AliyunTag `json:"tags"`
+	Page         int64       `json:"page"`
+	PageSize     int64       `json:"pageSize"`
+	TotalRecords int64       `json:"total"`
+}
+
+func (t *AliyunTag) GetName() string {
+	return t.Tag
+}
+
+func (t *AliyunTag) GetImageID() string {
+	return t.ImageID
+}
+
+func (t *AliyunTag) GetDigest() string {
+	return t.Digest
+}
+
+func (t *AliyunTag) GetSize() int64 {
+	return t.ImageSize
+}
+
+func (r *Registry) GetAliyunTagsWithDetails(image string) (tags []QuayTag, err error) {
+	crClient, err := cr.NewClientWithAccessKey("cn-shanghai", r.accessKey, r.accessSecret)
```

Comment:
`cn-shanghai` should be in a `key.go` as it's a pretty important magic string here :D

Reply:
This will be parameterized instead